packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls.tsx

```diff
@@ -24,13 +26,37 @@ export const Title = (): React.JSX.Element => {
   return <TitleElement>{title}</TitleElement>;
 };
 
+const LocationDetailViewRefresh = () => {
+  const [{ history }] = useControl({
+    type: 'NAVIGATE',
+  });
+
+  const [{ data, isLoading }, handleList] = useAction({
+    type: 'LIST_LOCATION_ITEMS',
+  });
+
+  const prefix = history.join('');
+
+  return (
+    <Refresh
+      disabled={isLoading || data.result.length <= 0}
+      onClick={() =>
+        handleList({ prefix, options: { refresh: true, pageSize: 1000 } })
```

Comment:
Should we be hardcoding the pageSize here?

Reply:
Yea at the moment I think that's what we decided to do for this. Originally there was a task to make these calls exhaustive, but we decided to just make the pageSize: 1000 for now

Reply:
We'll need to address when adding pagination